### PR TITLE
add linting

### PR DESCRIPTION
**Pull request by ledhed2222**
this is a MASSIVE PR, I know.

What i've done here is add our `@xrplf/eslint-config/base` and `@xrplf/prettier-config` to this repo and resolved 100% of linter errors and warnings. This meant I needed to refactor a lot of stuff, which does IMO result in the package being significantly better. pretty risky PR but I think we can publish this as a beta release and have some folks test on it.

### CODE REVIEW

File: src/enums/index.ts [hunk 1]
```diff
@@ -1,156 +1,10 @@
...
-);
-const TransactionResult = new BytesLookup(
-  enums.TRANSACTION_RESULTS,
-  TRANSACTION_RESULT_WIDTH
-);
-const Field = new FieldLookup(enums.FIELDS as Array<[string, FieldInfo]>);
-
+export { default as Field } from './Field'
+export { default as FieldInstance } from './FieldInstance'
+/* eslint-disable import/no-unused-modules --
```
Comment (ledhed2222) [inline src/enums/index.ts:3]: check out how much this file improved!

**Comment by ledhed2222** [comment]
NOTE - generally I tried to not touch the tests other than simple formatting changes and ensuring that changes made to source are reflected in terms of functionality in the tests. The coverage of the tests SHOULD not have changed here.

File: src/ShaMap/ShaMap.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import ShaMapInner from './ShaMapInner'
```
Comment (mvadari) [inline src/ShaMap/ShaMap.ts:1]: This file/folder name should be camelCase

File: src/ShaMap/ShaMapInner.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
+/* eslint-disable no-bitwise -- this file needs to use bitwise operations */
+import * as assert from 'assert'
```
Comment (mvadari) [inline src/ShaMap/ShaMapInner.ts:2]: ditto

File: src/enums/Bytes.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+/* eslint-disable no-bitwise -- this file needs to use bitwise operations */
+import { Buffer } from 'buffer/'
```
Comment (mvadari) [inline src/enums/Bytes.ts:2]: filename should be camelCase

**Review by mvadari — COMMENTED** [review]
Lots of files need to be camelCase, instead of being capitalized

**Comment by natenichols** [comment]
So, other than the comment about the `snake-case.ts` to `camelCase.ts` type filenames, I don't have a lot to comment on.

Excited about the changes in this PR, especially changes to abstractions over `definitions.json`. Definitely cleans up some rough edges.

**Pull request closed without merging**